xmodule/assets/library_content/public/js/library_content_edit_helpers.js

```diff
@@ -0,0 +1,54 @@
+/* JavaScript for special editing operations that can be done on LibraryContentXBlock */
+// This is a temporary UI improvements that will be removed when V2 content libraries became
+// fully functional
+
+/**
+ * Toggle the "Problem Type" settings section depending on selected library type.
+ * As for now, the V2 libraries don't support different problem types, so they can't be
+ * filtered by it. We're hiding the Problem Type field for them.
+ */
+function checkProblemTypeShouldBeVisible(editor) {
+    var libraries = editor.find('.wrapper-comp-settings.metadata_edit.is-active')
+        .data().metadata.source_library_id.options;
+    var selectedIndex = $("select[name='Library']", editor)[0].selectedIndex;
+    var libraryKey = libraries[selectedIndex].value;
+    var url = URI('/xblock')
+        .segment(editor.find('.xblock.xblock-studio_view.xblock-studio_view-library_content.xblock-initialized')
+        .data('usage-id'))
+        .segment('handler')
+        .segment('is_v2_library');
```

Comment:
from @bradenmacdonald :  

> 
> **Nit**: this is an implicit dependency on a third-party `URI.js` library being present in the global scope.
> 
> Something like this would avoid the implicit dependency and is also easier to understand for people not familiar with `URI.js` (like me):
> ```js
> const usageId = editor.find('.xblock-studio_view-library_content').data('usage-id');
> const url = `/xblock/${usageId}/handler/is_v2_library`;
> ```
> I also think you can simplify the CSS selector, as shown.
> 
> If you do need a class to manipulate URLs, I would recommend using [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) and [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) which are web standard and don't increase the JS bundle size, instead of `URI.js`.